image_proc/src/image_proc.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __clang__
+// TODO(dirk-thomas) custom implementation until we can use libc++ 3.9
```

Comment:
The version of clang included with Ubuntu Bionic (target for Crystal) is v7. I think we can likely remove this section and test with clang/libc++.

Reply:
This functionality has actually been refactored into `rcpputils`, so I would suggest using that. https://github.com/ros2/rcpputils/blob/master/include/rcpputils/filesystem_helper.hpp

Supporting macOS still presents issues when using `std::filesystem`, so this is currently the suggested approach.